src/util/worker/mod.rs

```diff
@@ -254,14 +273,13 @@ fn poll<R, T, U>(
 
         keep_going = fill_task_batch(&rx, &mut batch, batch_size, timeout);
         if !batch.is_empty() {
-            counter.fetch_sub(batch.len(), Ordering::SeqCst);
-            WORKER_PENDING_TASK_VEC
-                .with_label_values(&[&name])
-                .sub(batch.len() as i64);
-            WORKER_HANDLED_TASK_VEC
-                .with_label_values(&[&name])
-                .inc_by(batch.len() as i64);
+            // batch will be cleared after `run_batch`, so we need to store its length
+            // before `run_batch`.
+            let batch_len = batch.len();
             runner.run_batch(&mut batch);
+            counter.fetch_sub(batch_len, Ordering::SeqCst);
```

Comment:
Why move these lines after `run_batch`.

Reply:
To keep sync with future_worker and future_pool. In these two modules, pending count is decreased only after a work is done, i.e. "pending" actually means "pending + working".